src/axom/spin/internal/linear_bvh/bvh_traverse.hpp

```diff
@@ -68,10 +68,13 @@ AXOM_HOST_DEVICE inline void bvh_traverse(
   int32 stackptr = 0;
   todo[stackptr] = BARRIER;
 
+  int32 found_leaf = 0;
   int32 current_node = 0;
+
   while(current_node != BARRIER)
   {
-    if(!leaf_node(current_node))
+    // Traverse until we hit a leaf node or the barrier.
+    while(!leaf_node(current_node))
     {
       BBoxType left_bin = inner_nodes[current_node + 0];
       BBoxType right_bin = inner_nodes[current_node + 1];
```

Comment:
I'm assuming we want an explicit copy for performance reasons, but double checking that we don't want the following:
```suggestion
      const BBoxType& left_bin = inner_nodes[current_node + 0];
      const BBoxType& right_bin = inner_nodes[current_node + 1];
```

Presumably, the performance might be different on CPUs and GPUs?

Reply:
Copying by value didn't have any measurable effect on CPU performance, but it made the GPU traversal kernels considerably faster by roughly 20-30%. 

My rough understanding for what's going on is that if we use const references, a few of the loads don't get queued up until they're actually needed, after the first bounding box check is run; this actually ends up stalling the dependent bounding box check on those memory load instructions. By loading the bounding boxes by value, all the loads are issued up front, and some of the loads for the second bounding box can complete while we're checking the first bounding box.